### DISCUSSION TITLE
How to specify different lifetimes for different job declarations

### DISCUSSION THREAD

**Discussion by NikiforovAll** [quartznet/quartznet#1132] [Q&A] [answered]
How can I define job DI lifetime? E.g. some jobs are scoped and scheduled on a timely basis and others could be a singleton.

Example: 

I'm using `UseMicrosoftDependencyInjectionScopedJobFactory` extension method from Quartz.Extensions.DependencyInjection.
I have a job that uses scoped dependency (e.g. typed HTTP client), another job is some sort of a result processor based on channel (System.Threading.Channels) registered as a singleton. Is it safe to mix jobs like that?

**Comment by lahma** [comment, marked as answer]
There was a [recent change ](https://github.com/quartznet/quartznet/issues/1120) (scheduled for 3.3) where jobs are no longer resolved from container at all. Basically jobs should always be considered as new'ed (transient lifetime). As job could run concurrently by multiple triggers it's never safe to have job as singleton.

So just have your dependencies injected to job via constructor, they can be singletons or transient as you wish, but job lifetime cannot be configured.

**Reply by NikiforovAll** [reply]
Thanks, I've decided to use HostedService for my singleton instance.  I don't need to control scheduling in my case, so it should be a good fit

**Comment by elementh** [comment]
As per the changes @lahma commented, if you really need a singleton job, you are better off abstracting that logic in a singleton service and calling it from inside the job.
